### PR TITLE
Get rid of extra form rerendering (close T681241, close T673188)

### PR DESCRIPTION
**Pull request by churkin**

### CODE REVIEW

File: js/ui/form/instance_storage.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
+var Guid = require("../../core/guid"),
+    each = require("../../core/utils/iterator").each;
```
Comment (jsdmitry) [inline js/ui/form/instance_storage.js]: Please, use "import" instead of "require".

File: js/ui/form/instance_storage.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    each(handler) {
+        each(this._storage, function(guid, { instance, item }) {
+            if(instance) {
+                handler(instance, item, guid);
+            }
+        });
+    };
+}
+
+module.exports = InstanceStorage;
```
Comment (lukyanovas) [inline js/ui/form/instance_storage.js]: You can use es6 export here

Comment (lukyanovas) [reply js/ui/form/instance_storage.js]: `module.exports` should be used only for public components

File: js/ui/form/instance_storage.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                resultInstance = instance;
+
+                return false;
+            }
+        });
+
+        return resultInstance;
+    }
+
+    clear() {
```
Comment (lukyanovas) [inline js/ui/form/instance_storage.js:23]: I suggest to lift up public interface upper than private `_find`

Comment (churkin) [reply js/ui/form/instance_storage.js:23]: I've looked to another classes, for example:
https://github.com/DevExpress/DevExtreme/blob/18_2/js/ui/accordion.js#L486
https://github.com/DevExpress/DevExtreme/blob/18_2/js/ui/autocomplete.js#L256
private methods go first

**Review by TikhonovaTatyana — COMMENTED** [review]
The method's name is OK

File: js/ui/form/instance_storage.js (hunk 2 above)
Comment (churkin) [reply js/ui/form/instance_storage.js]: Yes I can, but I'll have to import this module as 
`var module = require('....').default`
Not so good

File: js/ui/form/ui.form.js [hunk 4]
```diff
@@ -1532,9 +1548,13 @@ var Form = Widget.inherit({
             validationGroupConfig = ValidationEngine.getGroupConfig(validationGroup);
 
         validationGroupConfig && validationGroupConfig.reset();
-        each(this._editorInstancesByField, function(dataField, editor) {
-            editor.reset();
-            editor.option("isValid", true);
+        this._instanceStorage.each((instance, item) => {
```
Comment (DokaRus) [inline js/ui/form/ui.form.js]: This file will not be transpiling to the ES5(while testing) without using `import`

File: js/ui/form/ui.form.js [hunk 5]
```diff
@@ -1599,7 +1616,17 @@ var Form = Widget.inherit({
...
+        return this._instanceStorage.findByDataField(dataField) || this._instanceStorage.findByName(dataField);
+    },
+
+    /**
+     * @name dxFormMethods.getButton
+     * @publicName getButton(name)
+     * @param1 name:string
+     * @return any
+     */
+    getButton: function(name) {
```
Comment (DokaRus) [inline js/ui/form/ui.form.js:1630]: @TikhonovaTatyana What do you think, maybe better we'll add neutral `getWidget` method, and marked `getEditor` as deprecated in 18.2?

Comment (TikhonovaTatyana) [reply js/ui/form/ui.form.js:1630]: I like smth like getItemWidget more than 2 separate methods

Comment (TikhonovaTatyana) [reply js/ui/form/ui.form.js:1630]: I've talked with Roma, he likes the variant with getButton

**Review by lukyanovas — APPROVED** [review]

**Review by jsdmitry — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Review by TikhonovaTatyana — APPROVED** [review]

File: js/ui/form/ui.form.layout_manager.js [hunk 6]
```diff
@@ -884,6 +885,8 @@ var LayoutManager = Widget.inherit({
                 errors.log("E1035", e.message);
             }
         }
+
+        return editorInstance;
```
Comment (IgnatovDan) [inline js/ui/form/ui.form.layout_manager.js:889]: The 'editorInstance' variable is not initialized always and this method can return 'undefined'.
The method return value will be added to the '_instanceStorage' collection and there are unsafe calls to members of this collection: "instance.reset();", "instance.registerKeyHandler", the public "getEditor: function(dataField)" method will return this value as well.

**Pull request merged**